torchao/prototype/mx_formats/custom_cast.py

```diff
@@ -294,8 +287,8 @@ def triton_f4_to_scaled_bf16_kernel(
         s = tl.load(s_ptr + offsets_s, mask=mask_s)
 
         # create the scale in bf16
-        s_offset = s.to(tl.int16) - e8m0_exponent_bias
-        s_fp = libdevice.pow(2.0, s_offset).to(tl.bfloat16)
+        # S is already biased by 127, so we just have to shift it to align w/ bf16
+        s_fp = (s.to(tl.uint16) << 7).to(tl.bfloat16, bitcast=True)
```

Comment:
So seems like we do actuallly do some unscaled casts: https://github.com/pytorch/ao/blob/463a87274f196f7c6cc16f9761940e29b3d123db/test/prototype/mx_formats/test_custom_cast.py#L316
Are you cool w/ addign another config to the triton kernel to specify